src/client/views/select-wallet.ts

```diff
@@ -38,36 +39,58 @@ const flowSVG = `
 
 export class SelectWallet extends AbstractView {
 	render() {
-		// TODO only show the wallets that relate to the tokens available.
 
+		const evmOrOffChain = this.client.evmAvailable() || this.client.offChainAvailable();
+
+		const SafeConnectButton = this.client.safeConnectAvailable()
+			? `<button class="wallet-button-tn" data-wallet="SafeConnect" aria-label="Safe connect wallet button">
+			  ${safeConnectSVG}
+			  <p>Safe Connect</p>
+			</button>`
+			: ''
+
+		const browser = getBrowserData();
 		const MetaMaskButton =
-			typeof window.ethereum !== 'undefined'
+			(browser.anyMetamask && evmOrOffChain)
```

Comment:
It's probably more reliable to use window.ethereum rather than explicitly checking for metamask as other non-metamask providers may exists. i.e. does this break Brave browser wallet support?

Reply:
as for me  - TN will show inaccurate icons for brave, opera and lot of mobile wallets if it will display Metamask, while no Metamask installed.
Lets allow WalletConnect to interact with other wallets.

Reply:
This happens in alot of dapps though. For instance in alphawallet. Will this prevent injected provider from being used in alphawallet?

Reply:
no, it just hides MetaMask  icon for non-Metamask browsers. 
It doesnt disable use of windows.ethereum
user can connect ANY wallet (metamask too) through WalletConnect

Reply:
@oleggrib yes but in that case the user has to pair wallet connect. I think it's better to have both options, even if the icon is wrong. We can always fix the icon and title of the wallet in a later version. 

In essence, let's fix it later rather than removing existing functionality.

Reply:
I have added icons for multiple wallets, so if window.ethereum defined then appropriate icon will be displayed.
in brave browser: 
- if metamask installed and enabled then metamask icon appear
- if brave wallet enabled then Brave icon appear

also added icons for:
- MEW
- trustWallet
- statusWallet
- alphaWallet
- imToken

Reply:
@oleggrib you are a legend!